printer/printer.go

```diff
@@ -175,11 +175,21 @@ func (z *printBuf) doString(indentLevel int, printState uint8, n datamodel.Node)
 	//  Note: this can be somewhat overbearing -- for example, typed strings are going to get called out as `string<String>{"value"}`.
 	//   This is rather agonizingly verbose, but also accurate; I'm not sure if we'd want to elide information about typed-vs-untyped entirely.
 	if tn, ok := n.(schema.TypedNode); ok {
-		z.writeString(tn.Type().TypeKind().String())
+		var tnk schema.TypeKind
+		var tntName string
+		// Defensively check for nil node type
+		if tnt := tn.Type(); tnt == nil {
+			tntName = "?!nil"
+			tnk = schema.TypeKind_Invalid
+		} else {
+			tntName = tnt.Name()
+			tnk = tnt.TypeKind()
+		}
+		z.writeString(tnk.String())
 		z.writeString("<")
-		z.writeString(tn.Type().Name())
+		z.writeString(tntName)
 		z.writeString(">")
-		switch tn.Type().TypeKind() {
+		switch tnk {
 		case schema.TypeKind_Invalid:
 			z.writeString("{?!}")
```

Comment:
this is getting into odd territory, but I think it should be fine to fall through and let the data-model driven behavior still do whatever it can?

Reply:
... on second thought, this is so very far from relevance that I don't know if there's a reason to care one way or the other, and I don't want to make you rewrite the tests you already made. It's fine as is.

Reply:
That's what I was thinking; initially I thought introducing a special implementation for `Type` and using that when the original is `nil` would leave the `printer` logic cleaner and more consistent. But doing that now is probably an overkill. 

I checked the rest of the logic and I think we are good to let it follow on. Happy to change if you think we shouldn't.

Reply:
OK I take all my optimism back 🤦 nil type when printing content still causes trouble

Reply:
Fixed here: https://github.com/ipld/go-ipld-prime/pull/298 

Apologies